mdp/test/test_UnevenlySampledSFANode.py

```diff
@@ -0,0 +1,28 @@
+from mdp.nodes.sfa_nodes import SFANode, UnevenlySampledSFANode
+from mdp import numx, Node
+from mdp.test._tools import assert_array_almost_equal, decimal
+
+
+def test_UnevenlySampledSFANode1():
+    # generate data
+    x = numx.random.random((100000, 3))
+    # minimal noise on time increment
+    dt = (numx.random.rand(x.shape[0]-1)-.5)*0.01 + 1.
+
+    # initialize nodes
+    sfa = SFANode()
+    unsfa = UnevenlySampledSFANode()
+
+    # train sfa
+    sfa.train(x)
+    unsfa.train(x, dt)
+
+    # stop training and generate slow features
+    sfa.stop_training()
+    unsfa.stop_training()
+
+    if ((sfa.sf[0, 0] > 0) and (unsfa.sf[0, 0] < 0)) \
+            or ((sfa.sf[0, 0] < 0) and (unsfa.sf[0, 0] > 0)):
+        sfa.sf *= -1.
+
+    assert_array_almost_equal(unsfa.sf, sfa.sf, decimal-6)
```

Comment:
Another good test for evenly against unevenly sampled SFA occurred to me. Create some data, then define some of its values to be missing (e.g. at a number of random locations). Once use [numpy.interp](https://docs.scipy.org/doc/numpy/reference/generated/numpy.interp.html) to fill the wholes, on the other hand create an appropriate dt to fill the wholes. Then original SFA should yield the same result on the interpolated data like uneven SFA would yield on the original data with proper `dt`.

Reply:
Would not it only be close to the same result? As we still approximate by using the trapezoid rule as opposed to a step function.

I will implement this. Should we also invest the time to adapt the original SFA tests to work with this algorithm?

Reply:
If I understand it correctly, the result by Trapezoid rule only differ from the naive linear estimation by other weights at the boundary. Can we have kind of an ordinary-SFA-compatibility flag that adjusts this to yield exactly (up to floating point precision) the same result as original SFA in this case? IMO that would be the clearest and safest way to test it. It would also enable to transparently study the benefits of trapezoid rule vs ordinary SFA.

Reply:
The main benefit I see in the trapezoid method is the better [rate of convergence](https://en.wikipedia.org/wiki/Riemann_sum). Although that advantage might not come into effect much in practice. 

The tests now compare the results of both current covariance methods up to floating point precision. So given nothing goes wrong inside of UnSFA, it should be what you are describing.